### ISSUE TITLE
[W tensorflow/core/framework/op_kernel.cc:993] Failed precondition

### ISSUE THREAD

**Issue by luoxiaoliaolan** [tensorflow/tensorflow#13035]
I want to run the tensorflow-deeplab-resnet,and I prepared my own training data for it. But When I try to run train.py to train data, it always did not work. And the error are as  follows:
![Uploading error.jpg…]()

I tensorflow/stream_executor/dso_loader.cc:135] successfully opened CUDA library libcublas.so.8.0 locally
I tensorflow/stream_executor/dso_loader.cc:135] successfully opened CUDA library libcudnn.so.5 locally
I tensorflow/stream_executor/dso_loader.cc:135] successfully opened CUDA library libcufft.so.8.0 locally
I tensorflow/stream_executor/dso_loader.cc:135] successfully opened CUDA library libcuda.so.1 locally
I tensorflow/stream_executor/dso_loader.cc:135] successfully opened CUDA library libcurand.so.8.0 locally
Tensor("create_inputs/batch:0", shape=(2, 4000, 4000, 3), dtype=float32) sssssssssssss
Tensor("create_inputs/batch:1", shape=(2, 4000, 4000, 1), dtype=uint8) sssssssssss
W tensorflow/core/platform/cpu_feature_guard.cc:45] The TensorFlow library wasn't compiled to use SSE3 instructions, but these are available on your machine and could speed up CPU computations.
W tensorflow/core/platform/cpu_feature_guard.cc:45] The TensorFlow library wasn't compiled to use SSE4.1 instructions, but these are available on your machine and could speed up CPU computations.
W tensorflow/core/platform/cpu_feature_guard.cc:45] The TensorFlow library wasn't compiled to use SSE4.2 instructions, but these are available on your machine and could speed up CPU computations.
W tensorflow/core/platform/cpu_feature_guard.cc:45] The TensorFlow library wasn't compiled to use AVX instructions, but these are available on your machine and could speed up CPU computations.
W tensorflow/core/platform/cpu_feature_guard.cc:45] The TensorFlow library wasn't compiled to use AVX2 instructions, but these are available on your machine and could speed up CPU computations.
W tensorflow/core/platform/cpu_feature_guard.cc:45] The TensorFlow library wasn't compiled to use FMA instructions, but these are available on your machine and could speed up CPU computations.
I tensorflow/core/common_runtime/gpu/gpu_device.cc:885] Found device 0 with properties: 
name: Quadro M4000
major: 5 minor: 2 memoryClockRate (GHz) 0.7725
pciBusID 0000:03:00.0
Total memory: 7.92GiB
Free memory: 7.51GiB
I tensorflow/core/common_runtime/gpu/gpu_device.cc:906] DMA: 0 
I tensorflow/core/common_runtime/gpu/gpu_device.cc:916] 0:   Y 
I tensorflow/core/common_runtime/gpu/gpu_device.cc:975] Creating TensorFlow device (/gpu:0) -> (device: 0, name: Quadro M4000, pci bus id: 0000:03:00.0)
Restored model parameters from /home/precision/code/tensorflow-deeplab-resnet-master/deeplab_resnet.ckpt
W tensorflow/core/framework/op_kernel.cc:993] Failed precondition: /home/precision/code/tensorflow-deeplab-resnet-master/dataset/VOCdevkit
W tensorflow/core/framework/op_kernel.cc:993] Failed precondition: /home/precision/code/tensorflow-deeplab-resnet-master/dataset/VOCdevkit
W tensorflow/core/framework/op_kernel.cc:993] Out of range: FIFOQueue '_1_create_inputs/batch/fifo_queue' is closed and has insufficient elements (requested 2, current size 0)
     [[Node: create_inputs/batch = QueueDequeueManyV2[component_types=[DT_FLOAT, DT_UINT8], timeout_ms=-1, _device="/job:localhost/replica:0/task:0/cpu:0"](create_inputs/batch/fifo_queue, create_inputs/batch/n)]]
Traceback (most recent call last):
  File "train.py", line 258, in <module>
    main()
  File "train.py", line 247, in main
    loss_value, images, labels, preds, summary, _ = sess.run([reduced_loss, image_batch, label_batch, pred, total_summary, train_op], feed_dict=feed_dict)
  File "/home/precision/anaconda2/envs/tensorflow/lib/python2.7/site-packages/tensorflow/python/client/session.py", line 767, in run
    run_metadata_ptr)
  File "/home/precision/anaconda2/envs/tensorflow/lib/python2.7/site-packages/tensorflow/python/client/session.py", line 965, in _run
    feed_dict_string, options, run_metadata)
  File "/home/precision/anaconda2/envs/tensorflow/lib/python2.7/site-packages/tensorflow/python/client/session.py", line 1015, in _do_run
    target_list, options, run_metadata)
  File "/home/precision/anaconda2/envs/tensorflow/lib/python2.7/site-packages/tensorflow/python/client/session.py", line 1035, in _do_call
    raise type(e)(node_def, op, message)
tensorflow.python.framework.errors_impl.OutOfRangeError: FIFOQueue '_1_create_inputs/batch/fifo_queue' is closed and has insufficient elements (requested 2, current size 0)
     [[Node: create_inputs/batch = QueueDequeueManyV2[component_types=[DT_FLOAT, DT_UINT8], timeout_ms=-1, _device="/job:localhost/replica:0/task:0/cpu:0"](create_inputs/batch/fifo_queue, create_inputs/batch/n)]]

Caused by op u'create_inputs/batch', defined at:
  File "train.py", line 258, in <module>
    main()
  File "train.py", line 142, in main
    image_batch, label_batch = reader.dequeue(args.batch_size)
  File "/home/precision/code/tensorflow-deeplab-resnet-master/deeplab_resnet/image_reader.py", line 180, in dequeue
    num_elements)
  File "/home/precision/anaconda2/envs/tensorflow/lib/python2.7/site-packages/tensorflow/python/training/input.py", line 872, in batch
    name=name)
  File "/home/precision/anaconda2/envs/tensorflow/lib/python2.7/site-packages/tensorflow/python/training/input.py", line 667, in _batch
    dequeued = queue.dequeue_many(batch_size, name=name)
  File "/home/precision/anaconda2/envs/tensorflow/lib/python2.7/site-packages/tensorflow/python/ops/data_flow_ops.py", line 458, in dequeue_many
    self._queue_ref, n=n, component_types=self._dtypes, name=name)
  File "/home/precision/anaconda2/envs/tensorflow/lib/python2.7/site-packages/tensorflow/python/ops/gen_data_flow_ops.py", line 1310, in _queue_dequeue_many_v2
    timeout_ms=timeout_ms, name=name)
  File "/home/precision/anaconda2/envs/tensorflow/lib/python2.7/site-packages/tensorflow/python/framework/op_def_library.py", line 763, in apply_op
    op_def=op_def)
  File "/home/precision/anaconda2/envs/tensorflow/lib/python2.7/site-packages/tensorflow/python/framework/ops.py", line 2327, in create_op
    original_op=self._default_original_op, op_def=op_def)
  File "/home/precision/anaconda2/envs/tensorflow/lib/python2.7/site-packages/tensorflow/python/framework/ops.py", line 1226, in __init__
    self._traceback = _extract_stack()

OutOfRangeError (see above for traceback): FIFOQueue '_1_create_inputs/batch/fifo_queue' is closed and has insufficient elements (requested 2, current size 0)
     [[Node: create_inputs/batch = QueueDequeueManyV2[component_types=[DT_FLOAT, DT_UINT8], timeout_ms=-1, _device="/job:localhost/replica:0/task:0/cpu:0"](create_inputs/batch/fifo_queue, create_inputs/batch/n)]]

**[W tensorflow/core/framework/op_kernel.cc:993] Failed precondition**
I have written the path address for the training set explicitly，but the training program doesn't work properly.

Can you help me with this problem?
                                                                 Thanks a lot!

**Comment by reedwm** [comment]
The first error seems to be

```
'_1_create_inputs/batch/fifo_queue' is closed and has insufficient elements (requested 2, current size 0)
```

which indicates tensorflow-deeplab-resnet is trying to get elements from a queue after it is closed.

You should file an issue with tensorflow-deeplab-resnet, or even better, ask on StackOverflow.

**Comment by fastlater** [comment]
@luoxiaoliaolan  did u make it run? I am trying to train with my own dataset also. Actually, I picked one class (person) and trying to ignore the other classes. However, the label and prediction images resulted are displayed as black. Then, when I try to inference, the result is incorrect. When I train the model using the original number of classes, 21, the result is ok. I read all the possible issues from the forum of the project but now, the forum is closed and I cannot get the info of how to make it work for less than 21 classes.

**Comment by luoxiaoliaolan** [comment]
@fastlater  I'm very glad that we studied it together！！I encountered this problem for a long time, and I tried to slove it, but I failed. The key to the problem is that the train.py  is not able to read the training data into the queue. Maybe it is a bug of the lastest TensorFlow（v1.2） or it is a bug of the  tensorflow-deeplab-resnet , but its coder closed issue, and I can't read the Q&A.
I will continue to solve the problem.
